yarn-project/foundation/src/abi/abi.ts

```diff
@@ -254,7 +256,9 @@ export interface FunctionDebugMetadata {
 export function getFunctionDebugMetadata(abi: ContractAbi, functionName: string): FunctionDebugMetadata | undefined {
   const functionIndex = abi.functions.findIndex(f => f.name === functionName);
   if (abi.debug && functionIndex !== -1) {
-    const debugSymbols = abi.debug.debugSymbols[functionIndex];
```

Comment:
Does this only run if there's a need for debug data? (I ask as if so, the fact that json is somewhat inefficient doesn't matter much)

Reply:
It's run if you execute a function that you have stored debug metadata for (the idea is that only developers will generate ABIs with debug metadata) and only for the function being executed

Reply:
If we see that it hurts performance we can change it to a structure that parses it lazily when accessed but I don't see it being an issue for now